private/Metaprogram.h

```diff
@@ -0,0 +1,349 @@
+#pragma once
+
+#include <tuple>
+#include <assert.h>
+
+#include "TypeList.h"
+#include "TupleManipulation.h"
+
+#define PANKU_LIST(...) \
+   using PankuUserProvidedList = type_list<__VA_ARGS__>; \
+   using PankuProcessedList = PankuMetaprogram::type_list_extract_heads<PankuUserProvidedList>; \
+   using PankuUserClassList = PankuProcessedList::type; \
+   using PankuDependencies = PankuProcessedList::headless_type; \
+   using PankuPath = PankuMetaprogram::topological_sort<PankuUserClassList, PankuDependencies>::type; \
+   using PankuPathPointerised = PankuMetaprogram::type_list_pointerise<PankuPath>::type; \
+   using PankuClassTuple = convert<PankuPathPointerised, std::tuple>; \
+   extern Panku<PankuClassTuple> panku;
+
+#define PANKU_INSTANCE Panku<PankuClassTuple> panku;
+
+#define STANDALONE(ClassName) type_list<ClassName>
+#define DEPENDENCY(ClassName, ...) type_list<ClassName, ##__VA_ARGS__>
+
+using namespace TypeList;
```

Comment:
Yep, also, it shouldn't be "using" here... It's on a header. I'll spell it out for all of them.